ml/jml/feature_set.h

```diff
@@ -365,6 +365,9 @@ class Mutable_Feature_Set : public Feature_Set {
     virtual std::tuple<const Feature *, const float *, int, int, size_t>
     get_data(bool need_sorted = false) const
     {
+        if(features.size() == 0)
+            throw ML::Exception("features_type is empty!");
```

Comment:
I'm not sure that this should be an error.  In a sparse classifier, it's certainly possible that rows would have no data.